robot/servo.py

```diff
@@ -119,6 +138,34 @@ def serial(self):
         """Serial number of the board."""
         return self._serial
 
+    def direct_command(self, command_name, *args) -> List[str]:
+        """
+        Issue a command directly to the arduino.
+
+        :Example:
+        >>> # arrives on the arduino as "my-command 4"
+        >>> servo_board.direct_command('my-command', 4)
+        ["first line response from my command", "second line"]
+
+        The arguments to this method are bundled as a list and passed to robotd.
+        We expect to immediately get back a response message (as well as the
+        usual status blob) which contains either valid data from the arduino or
+        a description of the failure.
+        """
+        command = (command_name,) + args
+        response = self.send_and_receive({'command': command})['response']
+        status = response['status']
+
+        # consume the broadcast status
+        self.receive()
+
+        if status == 'ok':
+            return response['data']
+        else:
+            for cls in (CommandError, InvalidResponse):
+                if cls.__name__ == response['type']:
+                    raise cls(response['description'])
+
```

Comment:
What happens if we receive a response with an invalid `type`?

Reply:
good point; I'll make that error harder.

Reply:
023496d